queries/risk_free_batches.sql

```diff
@@ -0,0 +1,42 @@
+with interactions as (select selector,
+                             target,
+                             case
+                                 when selector in (
+                                                   '\x095ea7b3', -- approve
+                                                   '\x2e1a7d4d', -- withdraw
+                                                   '\xa9059cbb', -- transfer
+                                                   '\x23b872dd' -- transferFrom
+                                     ) then true
+                                 else false
+                                 end as risk_free
+                      from gnosis_protocol_v2."GPv2Settlement_evt_Interaction"
+                      where evt_block_time between '{{StartTime}}' and '{{EndTime}}'),
+
+     no_interactions as (select evt_tx_hash
+                         from gnosis_protocol_v2."GPv2Settlement_evt_Settlement"
+                         where evt_block_time between '{{StartTime}}' and '{{EndTime}}'
+                           and evt_tx_hash not in (select evt_tx_hash
+                                                   from gnosis_protocol_v2."GPv2Settlement_evt_Interaction"
+                                                   where evt_block_time between '{{StartTime}}' and '{{EndTime}}')),
+
+     batch_interaction_counts as (select s.evt_tx_hash,
+                                         count(*)                                          as num_interactions,
+                                         sum(case when risk_free = true then 1 else 0 end) as risk_free
+                                  from gnosis_protocol_v2."GPv2Settlement_evt_Settlement" s
+                                           inner join gnosis_protocol_v2."GPv2Settlement_evt_Interaction" i
+                                                      on s.evt_tx_hash = i.evt_tx_hash
+                                           inner join interactions i2
+                                                      on i.selector = i2.selector
+                                                          and i.target = i2.target
+                                  where s.evt_block_time between '{{StartTime}}' and '{{EndTime}}'
+                                  group by s.evt_tx_hash),
+
+     combined_results as (select *
```

Comment:
Nit: instead of result, you can make it more explicitly by calling it: combined_risk_free_batches